### PR TITLE
[Merged by Bors] - feat: subset of a charted space is open iff each image in charts is

### PR DESCRIPTION
**Pull request by grunweg**
From sphere-eversion; I just cleaned it up slightly and submitted it.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/ChartedSpace.lean [hunk 1]
```diff
@@ -622,6 +622,19 @@ theorem chart_target_mem_nhds (x : M) : (chartAt H x).target ∈ 𝓝 (chartAt H
   (chartAt H x).open_target.mem_nhds <| mem_chart_target H x
 #align chart_target_mem_nhds chart_target_mem_nhds
 
+variable (M) in
+@[simp]
+theorem iUnion_source_eq_univ : (⋃ x : M, (chartAt H x).source) = (univ : Set M) :=
+  eq_univ_iff_forall.mpr fun x ↦ mem_iUnion.mpr ⟨x, mem_chart_source H x⟩
+
+theorem isOpen_iff (s : Set M) :
+    IsOpen s ↔ ∀ x : M, IsOpen <| chartAt H x '' ((chartAt H x).source ∩ s) := by
```
Comment (urkud) [inline Mathlib/Geometry/Manifold/ChartedSpace.lean:631]: What do you think about adding the following lemma somewhere in `Topology.Basic`?

```lean
lemma isOpen_iff_of_cover {ι} {f : ι → Set X} (ho : ∀ i, IsOpen (f i)) (hU : (⋃ i, f i) = univ) {s : Set X} :
    IsOpen s ↔ ∀ i, IsOpen (f i ∩ s)
```

Comment (grunweg) [reply Mathlib/Geometry/Manifold/ChartedSpace.lean:631]: That sounds like a good idea; will do so (not today any more).

Comment (grunweg) [reply Mathlib/Geometry/Manifold/ChartedSpace.lean:631]: Used that lemma in ChartedSpace: is clearer, but not much shorter.

**Comment by urkud** [comment]
Thank you! I renamed lemmas because they were in the `_root_` namespace without `chart` in their names.
bors d+

**Comment by grunweg** [comment]
That's even nicer than my version; thanks for the review and the golf!
bors r+

**Comment by urkud** [comment]
Looks like an issue with Internet connection.
bors merge

**Comment by grunweg** [comment]
Let's try this again
bors r+

**Comment by jcommelin** [comment]
bors r-

Can you please merge master. There might be some minor issues with this PR vs master...

**Comment by jcommelin** [comment]
Sorry :facepalm: wrong PR. Meant #9764

bors r+

**Pull request closed without merging**